core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataReader.java

```diff
@@ -49,18 +49,19 @@
   /**
    * Creates an instance of {@link BlockWorkerDataReader}.
    *
-   * @param reader the block reader to read data from
-   * @param offset the offset
-   * @param len the length to read
-   * @param chunkSize the chunk size
+   * @param blockWorker the block worker to create data reader
+   * @param request the block read request to create block reader
    */
-  private BlockWorkerDataReader(BlockReader reader,
-      long offset, long len, long chunkSize) {
-    mReader = reader;
-    Preconditions.checkArgument(chunkSize > 0);
-    mPos = offset;
-    mEnd = Math.min(mReader.getLength(), offset + len);
-    mChunkSize = chunkSize;
+  private BlockWorkerDataReader(BlockWorker blockWorker,
+      BlockReadRequest request) throws IOException {
```

Comment:
we shall avoid throw exceptions in a constructor.
a constructor is meant to be something light weight.
If an exception is potentially thrown, move it to the factory method (that's what we did)